### PR TITLE
feat(wallet): Added Show Test Networks Pref

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Added a `Show test networks` pref in `brave://setting/wallet` and passed to `Wallet` UI

Resolves <https://github.com/brave/brave-browser/issues/21650>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/164307484-94895301-d273-4c0b-b2cd-43dff76a1715.mov

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 1]
```diff
@@ -512,6 +512,11 @@ void BraveWalletService::SetDefaultBaseCryptocurrency(
   }
 }
 
+void BraveWalletService::GetShowWalletTestNetworks(
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_service.cc:515]: Please add a test for this method being backed by a pref.

**Review by supermassive — CHANGES_REQUESTED** [review]

**Review by goodov — APPROVED** [review]
`chromium_src` lgtm

**Review by josheleonard — COMMENTED** [review]
TypeScript looks good to me 👍

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet/browser/brave_wallet_service.cc:515]: Test added!
https://github.com/brave/brave-core/pull/13098/files#diff-c114760829ee3614f7d0bcef91204dc45e57455128412f7b4f66704df7c24bc6R1084-R1090

Comment (supermassive) [reply components/brave_wallet/browser/brave_wallet_service.cc:515]: I meant it should cover case when changing pref changes service method outcome;)

<details>
  <summary>So please add smth like this</summary>
  
```
diff --git a/browser/brave_wallet/brave_wallet_service_unittest.cc b/browser/brave_wallet/brave_wallet_service_unittest.cc
index 93c5311959..7e686898b5 100644
--- a/browser/brave_wallet/brave_wallet_service_unittest.cc
+++ b/browser/brave_wallet/brave_wallet_service_unittest.cc
@@ -459,6 +459,18 @@ class BraveWalletServiceUnitTest : public testing::Test {
     return default_cryptocurrency;
   }
 
+  bool GetShowWalletTestNetworks() {
+    base::RunLoop run_loop;
+    bool show_wallet_test_networks;
+    service_->GetShowWalletTestNetworks(
+        base::BindLambdaForTesting([&](bool b) {
+          show_wallet_test_networks = b;
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+    return show_wallet_test_networks;
+  }
+
   void SimulateOnGetImportInfo(const std::string& new_password,
                                bool result,
                                const ImportInfo& info,
@@ -1084,7 +1096,10 @@ TEST_F(BraveWalletServiceUnitTest, GetAndSetDefaultBaseCryptocurrency) {
 TEST_F(BraveWalletServiceUnitTest, GetShowWalletTestNetworks) {
   // Default value for kShowWalletTestNetworks should be false
   EXPECT_FALSE(GetPrefs()->GetBoolean(kShowWalletTestNetworks));
-  EXPECT_FALSE(GetShowWalletTestNetworks(GetPrefs()));
+  EXPECT_FALSE(GetShowWalletTestNetworks());
+
+  GetPrefs()->SetBoolean(kShowWalletTestNetworks, true);
+  EXPECT_TRUE(GetShowWalletTestNetworks());
 }
 
 TEST_F(BraveWalletServiceUnitTest, EthAddRemoveSetUserAssetVisible) {

```
  
</details>

Marking as approved.

**Review by supermassive — APPROVED** [review]

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet/browser/brave_wallet_service.cc:515]: Gotcha!
Updated, thank you for the help! 😎

**Review by josheleonard — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification for `master` completed via https://github.com/brave/brave-core/pull/13390#issuecomment-1132476731.

**Pull request merged**